src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,31 @@
+package duke.task;
+
+public abstract class Task {
+    protected String description;
+    protected boolean isDone = false;
+
+    public Task(String description) {
+        this.description = description;
+    }
+
+    public void setDone() {
+        this.isDone = true;
+    }
+
+    public void setUndone() {
+        this.isDone = false;
+    }
+
+    @Override
+    public String toString() {
+        char marker = this.isDone ? 'X' : ' ';
+        return String.format("[%c] %s", marker, this.description);
+    }
+
+    /**
+     * A method to return the string output for the txt file.
+     *
+     * @return The string output for the txt file.
+     */
+    public abstract String fileOutput();
```

Comment:
Should this be named more descriptive of what it does? 
The javadocs description and name of `fileOutput` gave me the idea that this produces some output based on what is in the file. But from its implementation, it seems to give the file the input String to be written.

But I did understand your intentions from your javadoc return param so you did a good job writing it 😊!